From 8b95a77f8ed49e97beec2fee3b9c1a3ab5f12e15 Mon Sep 17 00:00:00 2001
From: Matthew Monaco <matthew.monaco@0x01b.net>
Date: Fri, 24 Jan 2014 11:23:01 -0700
Subject: [PATCH 246/267] pam_systemd: Ignore vtnr when seat != seat0

logind considers it an error for a seat other than seat0 to have a
non-zero vtnr for CreateSession

(cherry picked from commit d7353ef6095f5e7db63d9cc898c7134b64482550)
---
 src/login/pam-module.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/login/pam-module.c b/src/login/pam-module.c
index 973daf7..7ebb5d2 100644
--- a/src/login/pam-module.c
+++ b/src/login/pam-module.c
@@ -335,6 +335,12 @@ _public_ PAM_EXTERN int pam_sm_open_session(
                         get_seat_from_display(display, NULL, &vtnr);
         }
 
+        if (seat && !streq(seat, "seat0")) {
+                pam_syslog(handle, LOG_DEBUG,
+                      "Ignoring vtnr %d for %s which is not seat0", vtnr, seat);
+                vtnr = 0;
+        }
+
         if (!type)
                 type = !isempty(display) ? "x11" :
                         !isempty(tty) ? "tty" : "unspecified";
-- 
1.8.4.5

